src/neuroconv/datainterfaces/behavior/facemap/facemapdatainterface.py

```diff
@@ -0,0 +1,394 @@
+import subprocess
+import sys
+from pathlib import Path
+from typing import Literal, Optional
+
+import h5py
+import numpy as np
+from pynwb.base import TimeSeries
+from pynwb.behavior import EyeTracking, PupilTracking, SpatialSeries
+from pynwb.core import DynamicTableRegion
+from pynwb.file import NWBFile
+
+from neuroconv.utils.dict import DeepDict
+
+from ..video.video_utils import get_video_timestamps
+from ....basetemporalalignmentinterface import BaseTemporalAlignmentInterface
+from ....tools import get_module
+from ....utils import get_base_schema, get_schema_from_hdmf_class
+
+FilePathType = str | Path
+
+
+def install_package(package):
+    subprocess.check_call([sys.executable, "-m", "pip", "install", package])
+
+
+try:
+    from ndx_facemap_motionsvd import MotionSVDMasks, MotionSVDSeries
+except ImportError:
+    # TODO: to be change when ndx-facemap-motionsvd version on pip
+    install_package("git+https://github.com/catalystneuro/ndx-facemap-motionsvd.git@main")
+    from ndx_facemap_motionsvd import MotionSVDMasks, MotionSVDSeries
+
+
+class FacemapInterface(BaseTemporalAlignmentInterface):
+    display_name = "Facemap"
+    help = "Interface for Facemap output."
+
+    keywords = ["eye tracking"]
+
+    def __init__(
+        self,
+        mat_file_path: FilePathType,
+        video_file_path: FilePathType,
+        first_n_components: int = 500,
+        include_multivideo_SVD: bool = True,
+        verbose: bool = True,
+    ):
+        """
+        Load and prepare data for facemap.
+
+        Parameters
+        ----------
+        mat_file_path : string or Path
+            Path to the .mat file.
+        video_file_path : string or Path
+            Path to the .avi file.
+        first_n_components : int, default: 500
```

Comment:
Is `first_n_components` like a stub_test or something different? If the former, the convention we have is to make it part of the conversion options in the nwbfile kwargs.